js/utilities.js

```diff
@@ -300,85 +300,97 @@ var ERMrest = (function(module) {
-        }
+                // no title, name, term, label column: use id:text type
+                // Check for id column whose type should not be integer or serial
+                var idCol = table.columns.all().filter(function (c) {
+                    return ((c.name.toLowerCase() === "id") && (c.type.name.indexOf('serial') === -1) && (c.type.name.indexOf('int') === -1));
+                });
 
-        // no title, name, term, label column: use id:text type
-        // Check for id column whose type should not be integer or serial
-        var idCol = table.columns.all().filter(function (c) {
-            return ((c.name.toLowerCase() === "id") && (c.type.name.indexOf('serial') === -1) && (c.type.name.indexOf('int') === -1));
-        });
 
+                // no id:text, use the unique key
+                // If id column exists
+                if (idCol.length && typeof data[idCol[0].name] === 'string') {
 
-        // no id:text, use the unique key
-        // If id column exists
-        if (idCol.length && typeof data[idCol[0].name] === 'string') {
-            return idCol[0].formatvalue(data[idCol[0].name], { context: context });
-        }
+                    result = idCol[0].formatvalue(data[idCol[0].name], {context: context});
 
-        // Get the columns for shortestKey
-        var keyColumns = table.shortestKey;
+                } else {
 
-        // TODO this check needs to change. it is supposed to check if the table has a key or not
-        // if (keyColumns.length >= table.columns.length) {
-        //     return null;
-        // }
+                    // Get the columns for shortestKey
+                    var keyColumns = table.shortestKey;
 
-        var values = [];
+                    // TODO this check needs to change. it is supposed to check if the table has a key or not
+                    // if (keyColumns.length >= table.columns.length) {
+                    //     return null;
+                    // }
 
-        // Iterate over the keycolumns to get their formatted values for `row_name` context
-        keyColumns.forEach(function(c) {
-            var value = c.formatvalue(data[c.name], { context: context });
-            values.push(value);
-        });
+                    var values = [];
+
+                    // Iterate over the keycolumns to get their formatted values for `row_name` context
+                    keyColumns.forEach(function (c) {
+                        var value = c.formatvalue(data[c.name], {context: context});
+                        values.push(value);
+                    });
+
+                    /*
+                     * join all values by ':' to get the display_name
+                     * Eg: displayName for values=["12", "DNA results for human specimen"] would be
+                     * "12:DNA results for human specimen"
+                     */
+                    result = values.join(':');
+                }
+            }
+
+            template = "{{name}}";
```

Comment:
@shinyichen You might want to change this to `template = {{{name}}}`. The name shouldn't be escaped.